### PR TITLE
Update algorithm to accept formatted and non formatted numbers

### PR DESCRIPTION
**Pull request by franciscocardoso**
This PR updates the module to accept both formatted - `900-70-0000` - and non formatted - `900700000` -  ITINs.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,40 +1,59 @@
...
+    "taxpayer",
+    "validator"
   ],
-  "options": {
-    "mocha": "--compilers js:babel-register --recursive --require should"
+  "homepage": "https://github.com/seegno/itin-validator",
+  "bugs": "https://github.com/seegno/itin-validator/issues",
+  "license": "MIT",
+  "author": "Seegno",
+  "main": "dist/src/index.js",
```
Comment (ruimarinho) [inline package.json]: remove /index.js?

Comment (franciscocardoso) [inline package.json]: 👍

File: src/index.js [hunk 2]
```diff
@@ -1,43 +1,41 @@
...
  */
 
-function isValid(itin) {
-  return expression.test(itin);
+export function isValid(value) {
+  return expression.test(value);
 }
 
 /**
  * Masks the ITIN with "X" placeholders to protect sensitive data,
```
Comment (ruimarinho) [inline src/index.js]: Mask

File: src/index.js [hunk 3]
```diff
@@ -1,43 +1,41 @@
...
  * while keeping some of the original digits for contextual recognition.
  *
- * E.g. "123456789" -> "XXXXX6789"
+ * E.g. "123456789" -> "XXXXX6789", "123-45-6789" -> "XXX-XX-6789".
  */
 
 export function mask(itin) {
```
Comment (ruimarinho) [inline src/index.js]: value for consistency?

File: test/index_test.js [hunk 4]
```diff
@@ -3,88 +3,48 @@
...
 
-describe('isValidItin', () => {
-  describe('default()', () => {
-    it('should return `false` if itin is invalid', () => {
-      for (const itin of invalidNumbers) {
-        isValidItin(itin).should.be.false();
-      }
+describe('ItinValidator', () => {
+  describe('isValid()', () => {
+    it('should return false if number is invalid', () => {
```
Comment (ruimarinho) [inline test/index_test.js]: Add backticks around false?

File: test/index_test.js [hunk 5]
```diff
@@ -3,88 +3,48 @@
...
         should.fail();
       } catch (e) {
         e.should.be.instanceOf(Error);
         e.message.should.equal('Invalid Individual Taxpayer Identification Number');
       }
     });
+
+    it('should mask a valid value', () => {
+      mask(numbers.valid[0]).should.equal('XXX-XX-0000');
+      mask(numbers.valid[3], { strict: true }).should.equal('XXXXX0000');
```
Comment (ruimarinho) [inline test/index_test.js]: remove options

File: test/index_test.js [hunk 6]
```diff
@@ -3,88 +3,48 @@
...
 
         should.fail();
       } catch (e) {
         e.should.be.instanceOf(Error);
         e.message.should.equal('Invalid Individual Taxpayer Identification Number');
       }
     });
+
+    it('should mask a valid value', () => {
+      mask(numbers.valid[0]).should.equal('XXX-XX-0000');
```
Comment (ruimarinho) [inline test/index_test.js:46]: What about adding a test for a input that contains spaces too?

File: package.json [hunk 7]
```diff
@@ -1,40 +1,59 @@
...
+  "main": "dist/src",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/seegno/itin-validator.git"
   },
   "scripts": {
-    "build": "rm -rf dist/* && babel --copy-files src/ --out-dir dist/",
-    "changelog": "github_changelog_generator --no-issues",
+    "changelog": "github_changelog_generator --no-issues --header-label='# Changelog' --future-release=v$npm_config_future_release && sed -i '' -e :a -e '$d;N;2,4ba' -e 'P;D' CHANGELOG.md",
+    "cover": "babel-node node_modules/.bin/isparta cover --report html _mocha -- $npm_package_options_mocha",
```
Comment (ruimarinho) [inline package.json]: I think it's time to rename this to coverage.

**Comment by ruimarinho** [comment]
Proposed README changes:

```diff
diff --git i/README.md w/README.md
index 1941384..2c0d1d7 100644
--- i/README.md
+++ w/README.md
@@ -1,5 +1,5 @@
 # itin-validator
-Validate and mask U.S. Individual Taxpayer Identification Numbers.
+Validate and mask a U.S. Individual Taxpayer Identification Numbers (ITIN).
 
 ## Status
 [![npm version][npm-image]][npm-url] [![build status][travis-image]][travis-url]
@@ -19,7 +19,7 @@ This method validates if the given value is a valid `Individual Taxpayer Identif
 1. `value` _(*)_: The value to validate.
 
 #### Returns
-_(boolean)_:  Returns `true` if `value` is a valid Individual Taxpayer Identification Number, else `false`.
+_(boolean)_:  Returns whether the input value is a valid ITIN or not.
 
 #### Example
 
@@ -37,8 +37,6 @@ isValid('900700000');
 // => true
 ```
 
---------------------------------------------------------------------------------
-
 ### `mask(value)`
 This method will help you protect this sensitive piece of information by obfuscating some digits.
 
@@ -46,7 +44,7 @@ This method will help you protect this sensitive piece of information by obfusca
 1. `value` _(*)_: The value to mask.
 
 #### Returns
-_(string)_: Returns the masked value.
+_(string)_: Returns the masked value by replacing value certain digits by `X`.
 
 #### Example
 
@@ -64,8 +62,6 @@ mask('900700000');
 // => XXXXX0000
 ```
 
---------------------------------------------------------------------------------
-
 ## Tests
 To test using a local installation of `node.js`:
 
@@ -73,10 +69,10 @@ To test using a local installation of `node.js`:
 npm test
 ```
 
-To test using Docker exclusively (similarly to what is done in Travis CI):
+To test using Docker exclusively:
 
 ```sh
-npm run testdocker
+docker-compose run --rm sut
 ```
 
 ## Release
```

File: test/index_test.js (hunk 6 above)
Comment (franciscocardoso) [inline test/index_test.js:46]: 👍

**Pull request merged**